image-segmentation/workflow_DGXA100_40GB_4.yaml

```diff
@@ -0,0 +1,204 @@
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: WorkflowTemplate
+metadata:
+  name: mlperf-segmentation-dataset
+  labels: {}
+spec:
+  params: []
+  workspaces:
+    - name: pvc
+      description: PVC to store dataset
+  results: []
+  type: SeqPod
+  seqPod:
+    steps:
+      - name: download-and-preprocess-dataset
+        image: registry.tensorstack.cn/t9k/mlperf-nvidia:image_segmentation_preprocessing
+        resources:
+          limits:
+            cpu: 2
+            memory: 8Gi
+        script: |
+          #!/bin/bash
+          set -e
+
+          # set HTTP/HTTPS proxy if necessary
+          # export http_proxy=
+          # export https_proxy=
+
+          # in case of ephemeral-storage
+          mkdir $(workspaces.pvc.path)/KiTS19
+          cd $(workspaces.pvc.path)/KiTS19
+          mkdir raw-data-dir
+          mkdir data
+          cd raw-data-dir
+          git clone https://github.com/neheller/kits19
+          cd kits19
+          pip3 install -r requirements.txt
+          python3 -m starter_code.get_imaging
+
+          cd /workspace/unet3d
+          python preprocess_dataset.py --data_dir $(workspaces.pvc.path)/KiTS19/raw-data-dir/kits19/data --results_dir $(workspaces.pvc.path)/KiTS19/data
+
+          cd $(workspaces.pvc.path)/KiTS19
+          rm -rf raw-data-dir
+
+---
+
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: WorkflowTemplate
+metadata:
+  name: mlperf-segmentation-train
+spec:
+  params:
+    - default: mlperf
+      description: Name of the PVC that stores dataset
+      name: pvc_name
+  resource:
+    failureRules:
+      fieldSelector: status.phase==Failed
+    manifest: |
+      apiVersion: batch.tensorstack.dev/v1beta1
+      kind: MPIJob
+      metadata:
+        name: mlperf-segmentation
+      spec:
+        mca:
+          btl: ^openib
+          pml: ob1
+        mpiHome: /usr/local/mpi
+        ssh:
+          sshdPath: /usr/sbin/sshd
+        scheduler:
```

Comment:
Don't use  `scheduler` unless a user chooses so. For convenience, we can keep the scheduler settings as  inline comments.

Reply:
modified